src/elements/content-sidebar/additional-tabs/AdditionalTabs.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * @flow
+ * @file Preview sidebar additional tabs component
+ * @author Box
+ */
+
+import React, { PureComponent } from 'react';
+import AdditionalTab from './AdditionalTab';
+import AdditionalTabsLoading from './AdditionalTabsLoading';
+
+import './AdditionalTabs.scss';
+
+type Props = {
+    tabs: ?Array<AdditionalSidebarTab>,
+};
+
+type State = {
+    isLoading: boolean,
+};
+
+class AdditionalTabs extends PureComponent<Props, State> {
+    numLoadedTabs: number = 0;
+
+    initialState: State = {
+        isLoading: true,
+    };
+
+    constructor(props: Props) {
+        super(props);
+
+        this.state = { ...this.initialState };
+    }
+
+    componentDidUpdate() {
+        const { tabs } = this.props;
+
+        if (tabs) {
+            // If we're displaying the more options tab, consider it loaded since it doesn't use an image.
+            const hasMoreTab = tabs.find(tab => tab.id < 0 && !tab.iconUrl);
```

Comment:
This logic being duplicated here and AdditionalTab makes it seem like the "more apps" button should be a separate component, instead.